### PR TITLE
code_80194710 OK (Audio libultra_code_O2)

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

This file is a simple copy-and-paste from OoT's:

https://github.com/zeldaret/oot/blob/master/src/libultra_code_O2/code_800E6840.c

It's notable that this file is put into the `libultra_code_O2` directory in OoT. There was no equivalent libultra structure in MM's folder-structure so it remains in the `src/code/audio/` for now. In spec, this file is also placed among all other audio files in `src/code/audio/`, but if there's a better place to move this file, let me know and I'll do it!

### CODE REVIEW

**Review by isghj5 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

File: include/functions.h [hunk 1]
```diff
@@ -3696,8 +3696,8 @@ AudioTask* func_80192BE0(void);
 // void func_80194568(void);
 // void func_80194668(void);
 // void func_801946E4(void);
-// void func_80194710(void);
-// void func_80194750(void);
+// void Audio_osInvalDCache(void);
+// void Audio_osWritebackDCache(void);
```
Comment (louist103) [inline include/functions.h:3700]: These should either be taken out or given an actual prototype.

**Review by EllipticEllipsis — APPROVED** [review]

**Comment by engineer124** [comment]
These two function names in this file were renamed in the OoT version from a PR earlier today. I am updating the names here to match how it was changed in OoT. The name changes are:
`Audio_osInvalDCache` -> `Audio_InvalDCache`
`Audio_osWritebackDCache` -> `Audio_WritebackDCache`

**Review by hensldm — APPROVED** [review]

**Comment by Kenix3** [comment]
Should we name this file something like "z_audio_dcache?"

**Comment by engineer124** [comment]
> Should we name this file something like "z_audio_dcache?"

The format for the audio files do not currently have the `z_` prefix. So if that's something we wanted to include, we'd have to change it for all of them. I'm fine with naming it `audio_dcache` for now

File: src/code/audio/audio_dcache.c [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#include "global.h"
+
+void Audio_InvalDCache(void* buf, s32 size) {
```
Comment (Kenix3) [inline src/code/audio/audio_dcache.c:3]: Hmm so this is a cache size. Thinking size_t is the appropriate type here. Anyone else have any input?

```suggestion
void Audio_InvalDCache(void* buf, size_t size) {
```

File: src/code/audio/audio_dcache.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    OSIntMask prevMask = osSetIntMask(1);
+
+    osInvalDCache(buf, size);
+    osSetIntMask(prevMask);
+}
+
+void Audio_WritebackDCache(void* buf, s32 size) {
```
Comment (Kenix3) [inline src/code/audio/audio_dcache.c:10]: Hmm so this is a cache size. Thinking size_t is the appropriate type here. Anyone else have any input?

```suggestion
void Audio_WritebackDCache(void* buf, size_t size) {
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]

File: src/code/audio/audio_dcache.c (hunk 3 above)
Comment (engineer124) [reply src/code/audio/audio_dcache.c]: Agreed, it's been changed for both functions. It's also being changed in an open PR for the OoT equivalent.

**Review by Kenix3 — APPROVED** [review]

**Review by louist103 — APPROVED** [review]

**Pull request merged**